server/db.js

```diff
@@ -87,7 +87,7 @@ const updateSong = (songdata) => {
   const { id, songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl } = songdata;
 
   return new Promise((resolve, reject) => {
-    const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8) where id = $1';
+    const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8, $9) where id = $1';
```

Comment:
Could you explain the syntax here? In particular, I'd like to know about what lead you to define a function inside the Promise and then invoke it within the promise.

Bear in mind, I'm not entirely familiar with the client.release() function or using 'try'